[#308] Add notification messages for image bulk upload completion

muloem opened the pull request.

muloem committed:
[#308] Add notification messages for image bulk upload completion

tangrammer commented on backend/src/akvo/flow_services/uploader.clj:
```diff
@@ -262,6 +262,18 @@
+        success-msg (partial (format "File: '%1$s' processed. '%4$s' out of '%3$s' images successfully added to instance id: '%2$s'") file-name form-instance-id total-images)
+        failure-msg (partial (format "Failed to add '%4$s' out of '%2$s' image files for folder '%3$s' in file '%1$s': %5$s ") file-name total-images form-instance-id)]
+    (doseq [response grouped-responses]
+      (if (= (:status response) 200)
```
```clojure
(doseq [[k v :as x] {1 ["a"], 2 ["as" "aa"], 3 ["asd"], 4 ["asdf" "qwer"]}]
  (println k v x))

```
output
```clojure
1 [a] [1 [a]]
2 [as aa] [2 [as aa]]
3 [asd] [3 [asd]]
4 [asdf qwer] [4 [asdf qwer]]
```
If you use doseq with a [group-by](https://clojuredocs.org/clojure.core/group-by) structure you don't have keys available (as `:status`  in each iteration)

Also using [doseq](https://clojuredocs.org/clojure.core/doseq) returns nil 

proposing in a new commit mapv instead (that is also eager)

tangrammer committed:
[#308] Refactor and fix use one use of doseq with group-by structure

https://github.com/akvo/akvo-flow-services/pull/313/files#diff-3a8debcd8296a12b786c774f0d2042a60632fa860e0c9d40f07bdb0ed5353128R272

tangrammer committed:
[#308] Log bulk image result

Thus using mapv instead of doseq allow us to return values between our
function calls

tangrammer reviewed (approved):
@muloem I've changed (and also added a fix) a bit the code to be in a more functional/clojure style ... I'll approve it now but instead of merging  I'll let you review it again and just in case merge

muloem committed:
[#308] update success message notifications

muloem committed:
[#308] Change signature for add-message function

* Fix logging when job complete
